Refactor/clean architecture

chiteshbansal opened the pull request:
1. Use the clean code architecture to refactor the code 
2. Directory structure updated 
3. More generic and common route handling

chiteshbansal committed:
TestCreateUser created for testing createUser controller

chiteshbansal committed:
Modified routes ,main and controller to handle the unit test function

chitesh-razorpay committed:
Unit test for create User controller using mock

chitesh-razorpay committed:
Unit Test function for GetUsers controller

chitesh-razorpay committed:
Test function for the getUserById controller

chitesh-razorpay committed:
Named Packages following the convention

chitesh-razorpay committed:
Added Configuration using viper

chitesh-razorpay committed:
Refactored and structured the code

anartify commented on .env:
```diff
@@ -0,0 +1,5 @@
+PORT = "3306"
```
Add the files you dont want to push in .gitignore file

chiteshbansal committed:
Delete .env

chiteshbansal committed:
Delete template.go

chitesh-razorpay committed:
1.Created gitignore file
2.Removed validator as a user method and created a common middleware and added to all user routes for validation

chitesh-razorpay committed:
Merge branch 'refactor/clean-architecture' of https://github.com/chiteshbansal/Basic-crud-apis-golang into refactor/clean-architecture

chiteshbansal committed:
Documented the code

chiteshbansal committed:
Created the generate jwt and verify jwt handlers

chitesh-razorpay committed:
Integrated the auth middleare with the api's and added the login functionaliy in the flow

chitesh-razorpay committed:
Created tests for the controllers
